### DISCUSSION TITLE
mpfs-hal-ddr-demo & getting DDR testing to pass

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#348] [Bare metal embedded software]
trying to executed the mpfs-hal-ddr-demo on both the Icicle Kit and the SEV-Kit.   I can changed configurations and build and load for either target, I just never get the DDR setup to complete.   execution just stays in the loop that looks like..

  while((ddr_status & DDR_SETUP_DONE) != DDR_SETUP_DONE)
    {
        ddr_status = ddr_state_machine(DDR_SS_MONITOR);
        next_time = report_status_functions(DDR_BOOT_PROGRESS, next_time);
    }

The behavior feels like DDR training is not done hence the DDR testing never succeeds, just hoping others have been down this path

Thx!
Dave

**Comment by mallynch** [comment]
Hi Dave,   I assume you are using SoftConsole to build the Demo. As a staring point, can you check you are using the correct build configuration.  You should select Icicle-Kit-LIM-Debug build configuration when debugging the Icicle kit. Thanks

**Reply by mallynch** [reply]
Hi Dave,   As a next step, can you make sure you program the board to be in mode0 before you start debugging. You only need to do this one. Mode zero0 insures the board is waiting for a debug session on power-up, as opposed to trying to run a program from eNVM. To put the board in mode0, from SoftConsole use the following menu : <run -> External tools -> PolarFire SoC program -idle boot mode 0>
Thanks

**Comment by rockoguido** [comment]
Hello.

Thanks for the quick response!
Yes I am building in SoftConsole and of course loading the image to the target via SoftConsole.   I also checked that I am using the correct build configuration.    As a test if I deviate from the correct buid then the image does not even load correctly, that was just a quick sanity check.

Am I to assume that the DDR training occurs before executing this application on the target?

Thx!

Dave Campanella
Principal Embedded Solutions Engineer / FAE
Microchip Technology Inc.
165 Technology Dr
Irvine, CA 92618
+1 925 989 0569 (mobile)
+1 760 304 0172 (office)
***@***.******@***.***>

[Text  Description automatically generated]
<http://www.microchip.com/emasters>Visit our new learning site at www.microchip.com/mu<http://www.microchip.com/mu>
